File: C++/4-estruturas/2091.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

bool compara(long long int a, long long int b) {
    return a < b;
}

int main() {
    ios::sync_with_stdio(false); cin.tie(0);

    long long int n;
    while (cin >> n) {
        if (n == 0) {
            break;
        }

        vector<long long int> numeros(n, 0);
        for (long long int i = 0; i < n; i++) {
            cin >> numeros[i];
        }

        sort(numeros.begin(), numeros.end(), compara);

        long long int resp = 0;
        long long int j = 0;
        while (j < n) {
            if (j == n-1) {
                resp = numeros[j];
                break;
            }

            if ((numeros[j] ^ numeros[j+1]) == 0) {
                j += 1;
            } else {
                resp = numeros[j];
            }
            j += 1;
        }

        cout << resp << "\n";

    }
}

File: Python/4-estruturas/2091.py
import sys

while True:
    n = int(sys.stdin.readline())

    if n == 0:
        break

    numeros = [int(x) for x in sys.stdin.readline().split()]
    resp = 0
    numeros.sort()

    i = 0
    while i < n:
        if i == n-1:
            resp = numeros[i]
            break

        if numeros[i] ^ numeros[i+1] == 0:
            i += 1
        else:
            resp = numeros[i]
        i += 1

    sys.stdout.write(f"{resp}\n")
